dotnet-desktop-guide/net/winforms/compiler-messages/wfo2001.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Compiler Warning WFO2001
+description: Learn more about compiler warning WFO2001. `Task` is being passed to `InvokeAsync` without a cancellation token.
+ms.date: 03/11/2025
+f1_keywords:
+  - "WFO2001"
+helpviewer_keywords:
+  - "WFO2001"
+---
+
+# Compiler Warning WFO2001
+
+**Version introduced:** .NET 9
+
+> Task is being passed to InvokeAsync without a cancellation token
+
+Provide a cancellation token when returning a <xref:System.Threading.Tasks.Task> or <xref:System.Threading.Tasks.ValueTask> from <xref:System.Windows.Forms.Control.InvokeAsync*>, which ensures the delegate can be properly awaited and exceptions can be handled.
+
+You can ignore this error if your intention is to run an unsupervised task.
+
+## To correct this warning
```

Comment:
The repro case: 
Note that `this.` is required to repro this due to a bug that we are servicing right now
 private void button1_Click(object sender, EventArgs e)
 {
     this.InvokeAsync(callback: DoSomethingAsync);

     async Task DoSomethingAsync()
     {
             await Task.Delay(100);
     }
 }

Reply:
And the fixed code
```
        private void button1_Click(object sender, EventArgs e)
        {
            this.InvokeAsync(callback: DoSomething);

            void DoSomething()
            {
                Task.Delay(100);
            }
        }
```
If you can come up with a more meaningful example, please do :)